src/main/java/ua/okwine/productexpirationdate/service/SkuService.java

```diff
@@ -0,0 +1,57 @@
+package ua.okwine.productexpirationdate.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ua.okwine.productexpirationdate.entity.Sku;
+import ua.okwine.productexpirationdate.entity.Supplier;
+import ua.okwine.productexpirationdate.exceptions.NotExistedSupplierId;
+import ua.okwine.productexpirationdate.repository.SkuRepository;
+import ua.okwine.productexpirationdate.repository.SupplierRepository;
+import ua.okwine.productexpirationdate.rest.dto.SkuDTO;
+import ua.okwine.productexpirationdate.rest.dto.SkuWithSupplierIdDTO;
+import ua.okwine.productexpirationdate.rest.dto.mapper.SkuMapper;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class SkuService {
+
+    private final SupplierRepository supplierRepository;
+    private final SkuRepository skuRepository;
+    private final SkuMapper skuMapper;
+
+    public SkuDTO create(SkuWithSupplierIdDTO skuWithSupplierIdDTO) {
+        Sku skuEntity = skuMapper.toSku(skuWithSupplierIdDTO);
+        Supplier supplier = supplierRepository.findById(skuWithSupplierIdDTO.getSupplier()).orElseThrow();
+        skuEntity.setSupplier(supplier);
+        Sku saveSku = skuRepository.save(skuEntity);
+
+        return skuMapper.toSkuDTO(saveSku);
+    }
+
+    public SkuDTO update(UUID id, SkuWithSupplierIdDTO skuWithSupplierIdDTO) {
+        Sku existingSku = skuRepository.findById(id).orElseThrow();
```

Comment:
what will throws if id not exist? To my way of thinking it must be a custom exception instead runtime exception or NPE.

Reply:
rename existingSku to sku

Reply:
ok